server/index.js

```diff
@@ -11,52 +12,98 @@ app.use(express.urlencoded({ extended: true }));
 app.use(cors());
 app.use(express.static(path.join(__dirname, '..', 'dist')));
 
-app.get('/', (req, res) => {
-  console.log('GOT a GET');
+// ////////////////////////Get Requests////////////////////////////// //
+
+app.get('/products/dept/:dept', async (req, res) => {
+  helpers.getDept(req.params.dept, (err, results) => {
+    res.status(200).json(results);
+  });
 });
 
-app.get('/products/dept/:dept', (req, res) => {
-  let formattedDept = formatName(req.params.dept);
-  RecommendedItem.find({ department: formattedDept }, (err, results) => {
-    res.json(results);
+app.get('/products/brand/:brandName', async (req, res) => {
+  helpers.getBrands(req.params.brandName, (err, results) => {
+    res.status(200).json(results);
   });
 });
 
-app.get('/products/brand/:brandName', (req, res) => {
-  // escape certain characters from request url
-  let brandWords = req.params.brandName.split(/[,.\s-&amp]/);
-  RecommendedItem.find({
-    brand: {$regex: `^${brandWords.join('.*\s*')}$`, $options: 'i'}}, (err, results) => {
-    res.json(results);
+app.get('/products/price/min=:minPrice&max=:maxPrice', async (req, res) => {
+  helpers.getPrices(req.params.minPrice, req.params.maxPrice, (err, results) => {
+    res.status(200).json(results);
   });
 });
 
-app.get('/products/price/min=:minPrice&max=:maxPrice', (req, res) => {
-  RecommendedItem.find({
-    price: {
-      $gte: req.params.minPrice || 0,
-      $lte: req.params.maxPrice || 1000 }
-    }, (err, results) => {
-      res.json(results);
+app.get('/products/id/:productId', async (req, res) => {
+  helpers.getProduct(req.params.productId, (err, searchedProduct) => {
+    if (err) {
+      console.error(err);
+      res.status(503).send('Error Caught at findOne a callback');
```

Comment:
I think I need to look more into different error codes.

Reply:
Ha! I was just guessing here..